packages/recorder/src/main.tsx

```diff
@@ -27,7 +27,10 @@ export const Main: React.FC = ({
   const [mode, setMode] = React.useState<Mode>('none');
 
   window.playwrightSetMode = setMode;
-  window.playwrightSetSources = setSources;
+  window.playwrightSetSources = React.useCallback((sources: Source[]) => {
+    setSources(sources);
+    window.playwrightSourcesEchoForTest = sources;
+  }, []);
```

Comment:
```suggestion
  }, [setSources]);
```
don't we need this dependency?

Reply:
state setters should not be passed into the watches